### PR TITLE
Add log-level as command line parameter

### PR DESCRIPTION
**Pull request by mrvdb**
Adds support for setting log-level to command line parameters.

### CODE REVIEW

File: src/astroid.cc [hunk 1]
```diff
@@ -158,6 +163,16 @@ namespace Astroid {
...
+    } else if (vm.count ("log-level")) {
+      ustring llevel = vm["log-level"].as<ustring> ();
+
+      /* Not very elegant map for string to enum */
+      std::map<std::string, logging::trivial::severity_level> sevmap = boost::assign::map_list_of
+        ("trace", logging::trivial::trace)("debug", logging::trivial::debug)
+        ("info", logging::trivial::info)("warning", logging::trivial::warning)
+        ("error", logging::trivial::error)("fatal", logging::trivial::fatal);
+
+       logging::core::get()->set_filter (logging::trivial::severity >= sevmap[llevel]);
```
Comment (gauteh) [inline src/astroid.cc]: What if llevel is not one of the legal options?

File: src/astroid.cc [hunk 2]
```diff
@@ -158,6 +163,16 @@ namespace Astroid {
...
       LOG (warn) << "disabling log";
       logging::core::get()->set_logging_enabled (false);
+    } else if (vm.count ("log-level")) {
+      ustring llevel = vm["log-level"].as<ustring> ();
+
+      /* Not very elegant map for string to enum */
+      std::map<std::string, logging::trivial::severity_level> sevmap = boost::assign::map_list_of
+        ("trace", logging::trivial::trace)("debug", logging::trivial::debug)
+        ("info", logging::trivial::info)("warning", logging::trivial::warning)
+        ("error", logging::trivial::error)("fatal", logging::trivial::fatal);
```
Comment (gauteh) [inline src/astroid.cc]: Is it possible to re-write this to only use standard C++14 or later? It's best to use Boost as little as possible, then we avoid the extra `include` as well.

File: src/astroid.cc (hunk 1 above)
Comment (mrvdb) [reply src/astroid.cc]: it will be silently ignored

File: src/astroid.cc (hunk 2 above)
Comment (mrvdb) [reply src/astroid.cc]: probably possible, this was just the easiest I could come up with.

out of curiosity, why use boost as little as possible, isn't the goal of a library to be used. Is there a severe penalty involved?

**Comment by gauteh** [comment]
Marcel van der Boom writes on september 18, 2017 13:57:
> mrvdb commented on this pull request.
> 
> 
> 
>> @@ -158,6 +163,16 @@ namespace Astroid {
>      if (vm.count ("disable-log")) {
>        LOG (warn) << "disabling log";
>        logging::core::get()->set_logging_enabled (false);
> +    } else if (vm.count ("log-level")) {
> +      ustring llevel = vm["log-level"].as<ustring> ();
> +
> +      /* Not very elegant map for string to enum */
> +      std::map<std::string, logging::trivial::severity_level> sevmap = boost::assign::map_list_of
> +        ("trace", logging::trivial::trace)("debug", logging::trivial::debug)
> +        ("info", logging::trivial::info)("warning", logging::trivial::warning)
> +        ("error", logging::trivial::error)("fatal", logging::trivial::fatal);
> 
> probably possible, this was just the easiest I could come up with.
> 
> out of curiosity, why use boost as little as possible, isn't the goal of a library to be used. Is there a severe penalty involved?

boost is pretty modular, so the less we use the less we need it. 

Ideally, if something is available in the standard C++ library we use 
that. An auxilliary library is unlikely to beat that in terms of code 
size, performance, clarity, maintainability, and stability. If it is in 
standard library it is easier for the next developer who reads your code 
to understand it. Docs and behavior would correspond to standard C++ and 
not the quirks of a specific library.

If we somewhen manage to get rid of a boost dependency that would make 
astroid significantly less complex, the less we use it, the easier that is.

File: src/astroid.cc (hunk 1 above)
Comment (gauteh) [reply src/astroid.cc]: ok, maybe put in a comment

**Comment by gauteh** [comment]
Thanks, merged!

**Pull request closed without merging**